### ISSUE TITLE
Lua Editor: EntityEntity_VM class is not present in the Classes Reference window and cannot be used

### ISSUE THREAD

**Issue by LB-KacperBieniek** [o3de/o3de#16609]
**Describe the bug**
Going through the [Lua Scripting Workflow Tests](https://github.com/o3de/sig-content/blob/main/testing-guidance/workflow-tests/scripting/lua-scripting-tests.md), and more precisely [Area: Moving entity using Lua script](https://github.com/o3de/sig-content/blob/main/testing-guidance/workflow-tests/scripting/lua-scripting-tests.md#area-moving-entity-using-lua-script) we encountered an issue with the _Modify the Lua script so that it uses keyboard input to move the entity_ workflow. The script used in this workflow is using the _EntityEntity_VM_ class, but running the script as it is throws errors in the _Console_. Searching for this class in _Classes Reference_ window returns no results. The class itself can be found in `Gems/ScriptCanvas/Assets/TranslationAssets/Classes/EntityEntity_VM.names`.

To workaround this issue the `EntityEntity_VM.GetEntityForward` method was changed to `ScriptCanvas_EntityFunctions_GetEntityForward`, and `EntityEntity_VM.GetEntityRight` method was changed to `ScriptCanvas_EntityFunctions_GetEntityRight`.

Additionally, it is possible that more classes are missing and cannot be used, but lack of documentation where all classes are listed makes it impossible to compare.

The issue is active on both Windows and Linux. It also occurs on both Development and Stabilization/2310 branches.

**Prerequisites**
- A project with _Remote Tools Connection_ Gem enabled.

**Steps to reproduce**
1. Open _Lua Editor_ tool.
2. Click the _Target: None_ button and choose _Editor (id)_ from the dropdown.
3. Wait for the _Classes Reference_ window to be populated.
4. Search for _EntityEntity_VM_ class.

**Expected behavior**
The _EntityEntity_VM_ class can be found in _Classes Reference_ window. The script using this class executes properly.

**Actual behavior**
The _EntityEntity_VM_ class cannot be found in _Classes Reference_ window. The script using this class executes with errors.

**Screenshot**

![NoClass](https://github.com/o3de/o3de/assets/87059746/abf49ba4-5210-488f-a3e7-a6f19d76bd61)

**Found in Branch**
Development
Stabilization/2310

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
Development: [f3dafe9](https://github.com/o3de/o3de/commit/f3dafe99700d2c0565fa26ab6f999440d0ada48e)
Stabilization/2310: [a39ed15](https://github.com/o3de/o3de/commit/a39ed1504ed45eb319f3a55ebc0285497978f483)

**Desktop:**
 - Device: PC
 - OS: Linux
 - Version Ubuntu 22.04.2 LTS
 - CPU Ryzen 5600H
 - GPU NVidia RTX 3070
 - Memory 16GB

**Comment by lsemp3d** [comment]
EntityEntity_VM is a product of a compiled Script Canvas graph and shouldn't be available for use directly in Lua. I'm curious how it was discovered to be available?

**Comment by LB-KacperBieniek** [comment]
@LB-ArturZieba who created the workflows in [Lua Scripting Workflow Tests](https://github.com/o3de/sig-content/blob/main/testing-guidance/workflow-tests/scripting/lua-scripting-tests.md) is not available until next week. I'll notify him to take a look at this when he returns.

**Comment by LB-ArturZieba** [comment]
@lsemp3d Hi, I've looked through different sample .lua scripts I keep for occasional use, but could not pinpoint the origin of using EntityEntity_VM. I've also tried to look through older O3DE (and Lumberyard) guides, but did not find anything.

If the use of the workaround from the issue is acceptable, then I'll change the script in the workflow and the workflows itself appropriately. This will also mean that this issue can be closed.
